components/ConvertSteps/Converter.js

```diff
@@ -0,0 +1,195 @@
+import { useTokenDecimals } from 'lib/web3-contracts'
+
+function Converter({ toAnj, amountSource, amountRecipient, handleReturnHome }) {
+  const checkAllowance = useAllowance()
+  const openOrder = useOpenOrder()
+  const claimOrder = useClaimOrder()
+  const antDecimals = useTokenDecimals('ANT')
+  const anjDecimals = useTokenDecimals('ANJ')
+
+  const [stepperStage, setStepperStage] = useState('working')
+  const [currentStep, setCurrentStep] = useState(
+    toAnj ? 'approval' : 'buyOrder'
+  )
+  const [stepState, setStepState] = useState({
+    approval: {
+      status: 'waiting',
+      active: true,
+    },
+    buyOrder: {
+      status: 'waiting',
+      active: toAnj ? false : true,
+      hash: '',
+    },
+    claimOrder: {
+      status: 'waiting',
+      active: false,
+      hash: '',
+    },
+  })
+
+  const formattedFromAmount = formatUnits(amountSource, {
+    digits: toAnj ? antDecimals : anjDecimals,
+    truncateToDecimalPlace: 8,
+    commas: true,
+  })
+
+  const formattedToAmount = formatUnits(amountRecipient, {
+    digits: toAnj ? anjDecimals : antDecimals,
+    truncateToDecimalPlace: 8,
+    commas: true,
+  })
+
+  const applyStepState = useCallback((step, status, hash) => {
+    setStepState(prevState => {
+      return {
+        ...prevState,
+        [step]: {
+          status: status,
+          active: true,
+          hash: hash ? hash : prevState[step].hash,
+        },
+      }
+    })
+  }, [])
+
+  const handleClaimOrderStep = useCallback(
+    async buyOrderHash => {
+      const stepType = 'claimOrder'
+
+      setCurrentStep(stepType)
+
+      try {
+        // Awaiting confirmation
+        applyStepState(stepType, 'waiting')
+        const transaction = await claimOrder(buyOrderHash, toAnj)
+
+        // Mining transaction
+        applyStepState(stepType, 'working', transaction.hash)
+        await transaction.wait()
+
+        // Success
+        applyStepState(stepType, 'success')
+        setStepperStage('success')
+      } catch (err) {
+        applyStepState(stepType, 'error')
+        setStepperStage('error')
+        console.log(err)
+      }
+    },
+    [claimOrder, applyStepState, toAnj]
+  )
+
+  const handleBuyOrderStep = useCallback(async () => {
+    const stepType = 'buyOrder'
+
+    setCurrentStep(stepType)
+
+    try {
+      // Awaiting confirmation
+      applyStepState(stepType, 'waiting')
+      const transaction = await openOrder(amountSource, toAnj)
+
+      // Mining transaction
+      applyStepState(stepType, 'working', transaction.hash)
+      await transaction.wait()
+
+      // Success
+      applyStepState(stepType, 'success')
+      handleClaimOrderStep(transaction.hash)
+    } catch (err) {
+      applyStepState(stepType, 'error')
+      setStepperStage('error')
+      console.log(err)
+    }
+  }, [amountSource, openOrder, applyStepState, toAnj, handleClaimOrderStep])
+
+  const handleApprovalStep = useCallback(async () => {
```

Comment:
These three actions are more or less the same, just explicitly wired up right now (see main review comment).

We can abstract this transaction flow with these degrees of freedom:
- Arbitrary async function to do something (e.g. create a transaction)
- Arbitrary callbacks to let the "controller" know that the action has succeeded, failed, or was cancelled